qlib/data/ops.py

```diff
@@ -32,6 +32,91 @@
 
 np.seterr(invalid="ignore")
 
+
+#################### Change instrument ########################
+# In some case, one may want to change to another instrument when calculating, for example
+# calculate beta of a stock with respect to a market index
+# this would require change the calculation of features from the stock (original instrument) to
+# the index (reference instrument)
+# #############################
+
+
+class ChangeInstrument(ExpressionOps):
+    """Change Instrument Operator
+    In some case, one may want to change to another instrument when calculating, for example, to
+    calculate beta of a stock with respect to a market index.
+    This would require changing the calculation of features from the stock (original instrument) to
+    the index (reference instrument)
+    Parameters
+    ----------
+    instrument: new instrument for which the downstream operations should be performed upon.
+                i.e., SH000300 (CSI300 index), or ^GPSC (SP500 index).
+
+    feature: the feature to be calculated for the new instrument.
+    Returns
+    ----------
+    Expression
+        feature operation output
+    """
+
+    def __init__(self, instrument, feature):
+        self.instrument = instrument
+        self.feature = feature
+
+    def __str__(self):
+        return "{}({},{})".format(type(self).__name__, self.instrument, self.feature)
+
+    def load(self, instrument, start_index, end_index, freq):
```

Comment:
`super().load(self, instrument, start_index, end_index, *args)`
Will it better to call the method from super class directly?

Reply:
Please let me know how you think and I can revise it. Thanks!

Reply:
Yes. 
I think this is a more concise way  to implement.

> Another way to achieve the same caching performance may be to call super().load(self, self.instrument, start_index, end_index, *args). This should do the trick without changing the _load_internal function, right?

Reply:
@wan9c9
Hi, did you encounter any problems when implementing the concise version  ?